### ISSUE TITLE
Android File System

### ISSUE THREAD

**Issue by maluoi** [maluoi/StereoKit#367]
## Describe the feature
StereoKit supports loading from Assets on Android, but the FilePicker doesn't browse the Android file system properly. This may be as simple as finding the correct folders and exposing them in the picker UI, or it may require some deeper Android API integration.

Some tips over here from Xamarin: https://docs.microsoft.com/en-us/xamarin/android/platform/files/

And some notes from ClonedPuppy's research
> `System.Environment.SpecialFolder.MyDocuments` is always r/w access on Android though, and that's the recommended folder to use for internal app storage
> or `System.Environment.SpecialFolder.Personal`, which maps directly to the same path as the above on Android
> which is basically "app_root_folder/files"
> on Android, from my limited reading the last few days, all other platform folders which has to do with personal stuff, maps to /files

**Comment by maluoi** [comment]
Are you looking for a generic full file picker for Android? This still hasn't been fleshed out further yet.

If your scope is a bit more limited, you should be able to handle it pretty easily with some custom UI and the standard file system APIs. Since Android is way more limited, especially with permissions, it's probably better to use a smaller scope for file access anyhow?

**Comment by mrstruijk** [comment]
I don't think anything fully fleshed out is needed. Just being able to grab the model files from the project's Asset folder would be lovely. Does Android complain about permissions in our own folders too? It shouldn't, right?

**Comment by maluoi** [comment]
Yeah, grabbing the asset files is pretty easy! Assets on Android are a little weird, but here's something close to what you'd use to grab the asset files. I haven't tested this, so there might be quirks related to directory recursion, but:

```csharp
List<string> files;
#if ANDROID
files = new List<string>(Android.App.Application.Context.Assets.List(""));
#else
files = new List<string>(Directory.GetFiles(SK.Settings.assetsFolder));
#endif
```

**Comment by maluoi** [comment]
SK's built-in file picker isn't extendable in this fashion yet! I'd probably recommend just making a regular window that lists your assets :) A slider is interesting, but I might just go with a for loop of buttons! This also gives you a bit more control over how the items are displayed, if you wanted to have thumbnails, or 3D previews, or anything else like that.

Check out [the Asset Browser demo](https://github.com/StereoKit/StereoKit/blob/master/Examples/StereoKitTest/Demos/DemoAssets.cs), might be of interest to you for this sort of thing :)
![](https://stereokit.net/img/screenshots/Demos/AssetEnumeration.jpg)

**Comment by mrstruijk** [comment]
Thank you Nick, this is awesome! 

Might be useful to mark in the docs that the current FilePicker doesn't work on native Quest, with the above link in there as well :).
